pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -19,130 +17,119 @@ import (
 )
 
 func checkIfTrayInstalled() error {
-	/* We want to force installation whenever setup is ran
-	 * as we want the service config to point to the executable
-	 * with which the setup command was issued.
-	 */
-
-	return fmt.Errorf("Ignoring check and forcing installation of System Tray")
+	if os.FileExists(filepath.Join(constants.StartUpFolder, constants.TrayShortcutName)) {
+		return nil
+	}
+	return fmt.Errorf("CodeReady Containers tray is not Installed")
 }
 
-func fixTrayInstalled() error {
-	/* To avoid asking for elevated privileges again and again
-	 * this takes care of all the steps needed to have a running
-	 * tray by invoking a ps script which does the following:
-	 * a) Add logon as service permission for the user
-	 * b) Create the daemon service and start it
-	 * c) Add tray to startup folder and start it
-	 */
+func checkIfDaemonInstalled() error {
+	// We want to force the installation of daemon as we want to
+	// run the daemon from the crc executable which ran the setup
+	return fmt.Errorf("Ignoring check, forcing installation of daemon")
+}
 
-	tempDir, err := ioutil.TempDir("", "crc")
+func fixDaemonInstalled() error {
+	currentExecutablePath, err := goos.Executable()
 	if err != nil {
-		logging.Error("Failed creating temporary directory for tray installation")
-		return err
-	}
-
-	defer func() {
-		_ = goos.RemoveAll(tempDir)
-	}()
-
-	// prepare the ps script
-	binPath, err := goos.Executable()
+		return fmt.Errorf("Failed to find current executables path: %w", err)
+	}
+
+	// Create the Posh script to launch daemon in the background with hidden window
+	daemonLaunchPSscriptContent := fmt.Sprintf(`$psi = New-Object System.Diagnostics.ProcessStartInfo
+$newproc = New-Object System.Diagnostics.Process
+$psi.FileName = "%s"
+$psi.Arguments = "%s"
+$psi.CreateNoWindow = $true
+$psi.WindowStyle = 'Hidden'
+$newproc.StartInfo = $psi
+$newproc.Start()
+$newproc
+`, currentExecutablePath, "daemon --log-level debug")
+
+	err = ioutil.WriteFile(constants.DaemonPSscriptPath, []byte(daemonLaunchPSscriptContent), 0600)
 	if err != nil {
-		return fmt.Errorf("Unable to find the current executables location: %v", err)
+		return fmt.Errorf("Failed to create required PS script file for the daemon: %w", err)
```

Comment:
PS is PowerShell I think? I'd expand the acronym in the error message, unless it's very common terminology in Windows.

Reply:
Expand